warp/src/multipass/identity.rs

```diff
@@ -61,11 +63,26 @@ impl IdentityProfile {
         }
     }
 
+    #[cfg_attr(target_arch = "wasm32", wasm_bindgen::prelude::wasm_bindgen(js_name = identity))]
+    pub fn identity_wasm(&self) -> Identity {
+        self.identity.clone()
+    }
+
+    #[cfg_attr(target_arch = "wasm32", wasm_bindgen::prelude::wasm_bindgen(js_name = set_identity))]
+    pub fn set_identity_wasm(&mut self, identity: Identity) {
+        self.identity = identity;
+    }
+
+    #[cfg_attr(target_arch = "wasm32", wasm_bindgen::prelude::wasm_bindgen(js_name = passphrase))]
+    pub fn passphrase_wasm(&self) -> Option<String> {
+        self.passphrase.as_ref().map(|phrase| phrase.to_string())
+    }
+}
```

Comment:
here, what we could do instead is just split the impl between wasm32 and non-wasm32 so we dont have to worry about the naming of the function through wasm-bindgen

Reply:
Thanks, I split it now, however note that the function still can't have the original name. On wasm the altered signature is there for being callable from js. But we still need the original signature when it is called internally within the wasm by other parts of our code. Since we can't have 2 methods with the same name, we need to pick a different name for the one with the altered function signature, hence the `_wasm`. The wasm-bindgen renaming is only there to get rid of the `_wasm` on the js side since it is redundant.

Reply:
As I think about it, I overlooked that I was borrowing the `passphrase` field in `IdentityProfile::passphrase`.